crates/nostr/src/event/builder.rs

```diff
@@ -1503,6 +1503,69 @@ impl EventBuilder {
         Self::gift_wrap_from_seal(receiver, &seal, expiration)
     }
 
+    /// Gift Wrap from seal with tags
+    ///
+    /// <https://github.com/nostr-protocol/nips/blob/master/59.md>
+    #[cfg(all(feature = "std", feature = "nip59"))]
+    pub fn gift_wrap_from_seal_with_tags(
+        receiver: &PublicKey,
+        seal: &Event,
+        expiration: Option<Timestamp>,
+        extra_tags: Vec<Tag>,
+    ) -> Result<Event, Error> {
```

Comment:
I think we can remove the `expiration` arg since can be added with `extra_tags`. And instead of having `gift_wrap` and `gift_wrap_with_tags` maybe is better to have a single `gift_wrap` method that take the `extra_tags` argument. The same for the `gift_wrap_from_seal` and `gift_wrap_from_seal_with_tags` method. What do you think?

Reply:
Yeah, I'd originally wanted to do that but then went with different methods because I wasn't sure about how you'd want to do this. Happy to change it.